tests/AulaTests.hs

```diff
@@ -163,5 +166,11 @@ boundary mn mx = frequency
     , (98, choose (mn, mx))
     ]
 
+nonEmptyST :: Gen ST
+nonEmptyST = (cs :: String -> ST) . getNonEmpty <$> arb
+
 nonEmptyMarkdown :: Gen Document
-nonEmptyMarkdown = Markdown . (cs :: String -> ST) . getNonEmpty <$> arb
+nonEmptyMarkdown = Markdown <$> nonEmptyST
+
+validTopicDescription :: Gen Description
+validTopicDescription = Description . Text.take Constant.topicDescMaxLength <$> nonEmptyST
```

Comment:
````haskell
data Contract a b = Contract {
      gen :: Gen a
    , box :: a -> b 
    , post :: b -> Prop 
    }

test :: Contract a b -> Spec
test = ...
````
I have the gut feeling this construction forms a category, which at least has products.

Apply TDD, would mean the define those contract, and test them against the function, can lead us a framework where we easily can put up complex test cases.